src/common/Input.tsx

```diff
@@ -137,6 +156,30 @@ const Input: React.FC<InputProps> = (props) => {
           color: #d2d2d2;
           text-decoration: line-through;
         }
+        .password-input {
+          padding-right: calc(var(--font-size-regular) * 2.8);
+        }
+        .input-wrapper {
+          position: relative;
+        }
+        .visibility-toggle {
```

Comment:
The CSS is great! Only one (small and subtle) issue: as the CSS is done right now, there is a sliiiiight shift in the icon position when it goes from "eye" to "eye with slash". This is because font awesome icons aren't all the same size, and the horizontal position of your icon is determined by the icon size + the padding. When the slash is added, it also adds a couple pixels to the icon's right side and the eye gets visually shifted to the left by that amount.

To avoid this shift, we can make the icon be centered within the label, and have the label width be the same regardless of the size of the icon. This way when the slash comes in, the eye will stay on the center, and the slash pixels being added to the side won't move its position. 

There's various ways to do this, but here's what I suggest:

```css
.visibility-toggle {
    // ...your previous CSS
   
    // Only add the padding on top and bottom
    padding: 13px 0px;
    // Fix the label width
    width: 50px; 
    // Center the content of the label within it, horizontally and vertically.
    display: flex;
    align-items: center;
    justify-content: center;
}
``` 

you will also need to change the padding right of password input to be fixed at `50px`.